File: src/backend/app/schemas/asignaturas.py
from pydantic import BaseModel, ConfigDict

from app.schemas.grados import GradoOut


class AsignaturaOut(BaseModel):
    """Asignatura del catálogo — lectura por GET /asignaturas y /profesores/yo."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    codigo: str
    nombre: str
    ects: float
    caracter: str
    curso_plan: int
    grado: GradoOut

File: src/frontend/src/types/asignaturas.ts
export interface Asignatura {
  id: number;
  codigo: string;
  nombre: string;
  ects: number;
  caracter: 'OB' | 'OP' | 'FB';
  curso_plan: number;
  plan_estudios: string;
  facultad: string;
}
